metacat-connector-druid/src/main/java/com.netflix.metacat.connector.druid/DruidConnectorFactory.java

```diff
@@ -0,0 +1,116 @@
+/*
+ *  Copyright 2017 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package com.netflix.metacat.connector.druid;
+
+import com.netflix.metacat.common.server.connectors.ConnectorContext;
+import com.netflix.metacat.common.server.connectors.ConnectorDatabaseService;
+import com.netflix.metacat.common.server.connectors.ConnectorFactory;
+import com.netflix.metacat.common.server.connectors.ConnectorPartitionService;
+import com.netflix.metacat.common.server.connectors.ConnectorTableService;
+import com.netflix.metacat.connector.druid.configs.DruidConnectorConfig;
+import com.netflix.metacat.connector.druid.configs.DruidHttpClientConfig;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.StandardEnvironment;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Druid Connector Factory.
+ *
+ * @author zhenl
+ * @since 1.1.0
+ */
+public class DruidConnectorFactory implements ConnectorFactory {
+    private final String catalogName;
+    private AnnotationConfigApplicationContext ctx;
+
+    /**
+     * Constructor.
+     *
+     * @param catalogName      connector name. Also the catalog name.
+     * @param connectorContext connector config
+     */
+    DruidConnectorFactory(
+        final String catalogName,
+        final ConnectorContext connectorContext
+    ) {
+        this.catalogName = catalogName;
+        this.ctx = new AnnotationConfigApplicationContext();
```

Comment:
So at this point this spring connector context code is in at least two places and looks to me like it's begging to be moved to some sort of either utility method or base class for code sharing. You really are doing the same set of steps

1. Configure the connector context bean
2. Set up an environment
3. Register some config classes
4. Refresh

You should at least be able to move the first three steps into some method that takes those parameters as input so it can be shared.